Reapply Redis password

Alirexaa opened the pull request:
## Description

[Add Password To Redis (dotnet/aspire#464)](https://github.com/dotnet/aspire/pull/4642) has been reverted because of [[WebToolsE2E][Aspire] Using ‘azd up’ to deploy aspire starter with redis project fails with error: generating bicep from manifest: argument 1 cannot contain connection strings, secured parameters, or secret outputs. Use environment variables instead. (dotnet/aspire#7429)](https://github.com/dotnet/aspire/issues/7429).

This PR addresses the mentioned issue by changing the container entrypoint and providing the password from an environment variable. 

For more information see https://github.com/dotnet/aspire/issues/3838#issuecomment-2655175621

Fixes #3838

@eerhardt @davidfowl 

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [x] Yes
    - If yes, did you have an API Review for it?
      - [x] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [x] Yes
      - [ ] No
  - [ ] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [ ] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Is this introducing a breaking change?
      - [ ] Yes
        - Link to aspire-docs issue (please use this [`breaking-change` template](https://github.com/dotnet/docs-aspire/issues/new?template=04-breaking-change.yml)):
      - [ ] No
        - Link to aspire-docs issue (please use this [`doc-idea` template](https://github.com/dotnet/docs-aspire/issues/new?template=02-docs-request.yml)):
  - [ ] No

Alirexaa committed:
Revert "Revert redis password change (#7518)"

This reverts commit 6a9540cc35299b55f61d0020defed8a5f1fe56df.

Alirexaa committed:
checkout eventhubns.module.bicep and eventhubs aspire-manifest.json

Alirexaa committed:
Set password from envs

Alirexaa committed:
Add redis-command in single arg instead of multiple args

Alirexaa committed:
Update playground manifest

Alirexaa committed:
Update playground manifests

Alirexaa committed:
fix test

davidfowl commented on src/Aspire.Hosting.Redis/RedisBuilderExtensions.cs:
```diff
@@ -61,7 +91,57 @@ public static IResourceBuilder<RedisResource> AddRedis(this IDistributedApplicat
+                }
+            });
+
+            builder.WithAnnotation(new CommandLineArgsCallbackAnnotation(context =>
```
You can just do WithArgs here right?

Alirexaa replied:
I did.

Alirexaa committed:
Address PR feedback

Alirexaa committed:
Merge branch 'dotnet:main' into revert-7518-RevertRedisPassword

eerhardt committed:
Merge remote-tracking branch 'upstream/main' into revert-7518-RevertRedisPassword

eerhardt committed:
Respond to PR feedback

eerhardt committed:
Change Redis password generation to exclude special characters.

SE.Redis doesn't support commas in the password when parsing the connection string.

eerhardt reviewed (approved):
Thanks again for this contribution @Alirexaa! Nice work here.

I merged main and refactored the code a little bit to simplify it a bit.
But when deploying the app to ACA I found an issue that while Aspire doesn't use commas `,` in its generated passwords, `azd` will. So I updated the code to not allow special characters in the password.

This change LGTM - we should be able to finally fix this long standing security issue. Thanks again!

eerhardt committed:
Fix ACA bug with HostAndPort

eerhardt commented on src/Aspire.Hosting.Azure.AppContainers/AzureContainerAppsInfrastructure.cs:
```diff
@@ -608,6 +608,7 @@ BicepValue<string> GetHostValue(string? prefix = null, string? suffix = null)
                     EndpointProperty.Url => GetHostValue($"{scheme}://", suffix: isHttpIngress ? null : $":{port}"),
                     EndpointProperty.Host or EndpointProperty.IPV4Host => GetHostValue(),
                     EndpointProperty.Port => port.ToString(CultureInfo.InvariantCulture),
+                    EndpointProperty.HostAndPort=> GetHostValue(suffix: isHttpIngress ? null : $":{port}"),
```
FYI - @davidfowl - I found a bug that we missed here and fixed it so all the generate-manifests work again.

eerhardt replied:
I didn't know if I should switch off `isHttpIngress` or not. If you explicitly ask for `HostAndPort` and this is the httpIngress, should we still tack on `:80`?

eerhardt replied:
Rethinking this, I think it should always put the port. The reasoning the URL can drop the port is because it contains the scheme at the begining. `http://` indicates what the port is. If you say `HostAndPort` we should always give you the host and the port.

davidfowl replied:
I agree, it should include host an port

eerhardt committed:
Always display the port for HostAndPort

eerhardt commented on src/Aspire.Hosting.Azure.AppContainers/AzureContainerAppsInfrastructure.cs:
```diff
@@ -608,6 +608,7 @@ BicepValue<string> GetHostValue(string? prefix = null, string? suffix = null)
                     EndpointProperty.Url => GetHostValue($"{scheme}://", suffix: isHttpIngress ? null : $":{port}"),
                     EndpointProperty.Host or EndpointProperty.IPV4Host => GetHostValue(),
                     EndpointProperty.Port => port.ToString(CultureInfo.InvariantCulture),
+                    EndpointProperty.HostAndPort=> GetHostValue(suffix: $":{port}"),
```
```suggestion
                    EndpointProperty.HostAndPort => GetHostValue(suffix: $":{port}"),
```

eerhardt committed:
Update src/Aspire.Hosting.Azure.AppContainers/AzureContainerAppsInfrastructure.cs
